mlir/test/Target/LLVMIR/nvvmir.mlir

```diff
@@ -556,19 +556,16 @@ llvm.func @kernel_func() attributes {nvvm.kernel} {
   llvm.return
 }
 
-// CHECK:     !nvvm.annotations =
-// CHECK-NOT: {ptr @nvvm_special_regs, !"kernel", i32 1}
-// CHECK:     {ptr @kernel_func, !"kernel", i32 1}
+// CHECK: ptx_kernel void @kernel_func
```

Comment:
So, downstream (non-upstream ones) users should migrate to this cc-based instead of kernel-metadata based, right?
I am asking since this does not seem to be an opt-in/optional change.

Alex, Could you please clarify?

Reply:
This change does not remove support for specifying a kernel via the metadata. It simply updates frontends and tests to use a different one of the two already supported methods for marking kernels. Long term I hope to remove the support for metadata, so downstream users should move the calling-convention, but this change does not yet force that.

Reply:
ok, thanks for clarifying!